tests/Exception/RequestExceptionTest.php

```diff
@@ -152,4 +152,19 @@ public function testCanProvideHandlerContext()
         $e = new RequestException('foo', $r, null, null, ['bar' => 'baz']);
         $this->assertEquals(['bar' => 'baz'], $e->getHandlerContext());
     }
+
+    public function testObfuscateUrlWithUsername()
+    {
+        $r = new Request('GET', 'http://username@www.oo.com');
+        $e = RequestException::create($r, new Response(500));
+        $this->assertContains('username@', $e->getMessage());
+    }
+
+    public function testObfuscateUrlWithUsernameAndPassword()
+    {
+        $r = new Request('GET', 'http://user:password@www.oo.com');
+        $e = RequestException::create($r, new Response(500));
+        $this->assertNotContains('user:password', $e->getMessage());
```

Comment:
Testing if something is not in the URL can be omitted if you test the whole URL here too.